rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -0,0 +1,311 @@
+import logging
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.ethereum.utils import (
+    token_normalized_value,
+    token_normalized_value_decimals,
+)
+from rotkehlchen.chain.evm.constants import DEFAULT_TOKEN_DECIMALS, ZERO_ADDRESS
+from rotkehlchen.chain.evm.decoding.hop.constants import CPT_HOP, HOP_CPT_DETAILS
+from rotkehlchen.chain.evm.decoding.hop.structures import HopBridgeEventData
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    ActionItem,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.fval import FVal
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChainID, ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+from .constants import (
+    TOKEN_SWAP,
+    TRANSFER_FROM_L1_COMPLETED,
+    TRANSFER_SENT,
+    WITHDRAWAL_BONDED,
+    WITHDREW,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class HopCommonDecoder(DecoderInterface):
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            bridges: dict[ChecksumEvmAddress, HopBridgeEventData],
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.bridges = bridges
+
+    def _get_bridgeasset_amount(self, amount_raw: int, bridge: HopBridgeEventData) -> FVal:
+        """Normalize raw amount based on bridge asset type."""
+        if bridge.identifier == A_ETH.identifier:
+            return token_normalized_value_decimals(amount_raw, DEFAULT_TOKEN_DECIMALS)
+        return token_normalized_value(amount_raw, Asset(bridge.identifier).resolve_to_evm_token())
+
+    def _get_bridge_note(
```

Comment:
naming nitpick: you don't get it. You create/generate it

```suggestion
    def _generate_bridge_note(
```